addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxControlColorPickerV2.java

```diff
@@ -0,0 +1,294 @@
+     * @param toHigh   the upper bound of the value's target range
+     * @return the mapped value
+     */
+    private int map(int x, int fromLow, int fromHigh, int toLow, int toHigh) {
+        return (x - fromLow) * (toHigh - toLow) / (fromHigh - fromLow) + toLow;
+    }
+
+    /**
+     * Constrains a number to be within a range.
+     *
+     * @param x   the number to constrain
+     * @param min the minimum value
+     * @param max the maximum value
+     * @return the constrained value
+     */
+    private int constrain(int x, int min, int max) {
+        if (x >= min && x <= max) {
+            return x;
+        } else if (x < min) {
+            return min;
+        } else {
+            return max;
+        }
+    }
+
+    /**
+     * calculates the red value based on the Kelvin temperature
+     *
+     * @param temp the Kelvin temperature
+     * @return the red value
+     */
+    private int calculateRed(int temp) {
+        int red = 255;
+        int temperature = temp / 100;
+
+        if (temperature > 66) {
+            red = temperature - 60;
+            red = ((Long) Math.round(329.698727466 * Math.pow(red, -0.1332047592))).intValue();
+        }
+
+        return constrain(red, 0, 255);
+    }
+
+    /**
+     * calculates the green value based on the Kelvin temperature
+     *
+     * @param temp green Kelvin temperature
+     * @return the red value
+     */
+    private int calculateGreen(int temp) {
+        int green;
+        int temperature = temp / 100;
+
+        if (temperature <= 66) {
+            green = temperature;
+            green = ((Long) Math.round((99.4708025861 * Math.log(green)) - 161.1195681661)).intValue();
+        } else {
+            green = temperature - 60;
+            green = ((Long) Math.round(288.1221695283 * Math.pow(green, -0.0755148492))).intValue();
+        }
+
+        return constrain(green, 0, 255);
+    }
+
+    /**
+     * calculates the blue value based on the Kelvin temperature
+     *
+     * @param temp the Kelvin temperature
+     * @return the blue value
+     */
+    private int calculateBlue(int temp) {
+        int blue = 255;
+        int temperature = temp / 100;
+
+        if (temperature < 65) {
+            if (temperature <= 19) {
+                blue = 0;
+            } else {
+                blue = temperature - 10;
+                blue = ((Long) Math.round((138.5177312231 * Math.log(blue)) - 305.0447927307)).intValue();
+            }
+        }
+
+        return constrain(blue, 0, 255);
+    }
+
+    /**
+     * checks whether the given Loxone color is a temperature
+     *
+     * @param color the loxone color
+     * @return boolean
+     */
+    private boolean isTemperatureColor(String color) {
+        return color.startsWith(COLOR_TYPE_TEMP);
+    }
+
+    /**
+     * checks whether the given Loxone color is a HSV-color
+     *
+     * @param color the loxone color
+     * @return boolean
+     */
+    private boolean isHsvColor(String color) {
```

Comment:
It is a one liner and used only once, maybe it would make sense to just use it in the body of the calling method. Similar in is temperatureColor.